tools/common-tools/src/demos/sync/index.js

```diff
@@ -0,0 +1,220 @@
+import "./live-common-setup";
+import React, { useEffect, useState } from "react";
+import { BigNumber } from "bignumber.js";
+import {
+  encodeAccountId,
+  decodeAccountId,
+  shortAddressPreview,
+  emptyHistoryCache,
+} from "@ledgerhq/live-common/lib/account";
+import {
+  findCryptoCurrencyById,
+  getCryptoCurrencyById,
+} from "@ledgerhq/live-common/lib/currencies/index";
+import { getAccountBridge } from "@ledgerhq/live-common/lib/bridge";
+import { reduce } from "rxjs/operators";
+import { makeBridgeCacheSystem } from "@ledgerhq/live-common/lib/bridge/cache";
+import {
+  getDerivationScheme,
+  runDerivationScheme,
+  asDerivationMode,
+} from "@ledgerhq/live-common/lib/derivation";
+
+const localCache = {};
+const bridgeCache = makeBridgeCacheSystem({
+  saveData(c, d) {
+    localCache[c.id] = d;
+    return Promise.resolve();
+  },
+
+  getData(c) {
+    return Promise.resolve(localCache[c.id]);
+  },
+});
+
+function inferAccount(id) {
+  const { derivationMode, xpubOrAddress, currencyId } = decodeAccountId(id);
+  const currency = getCryptoCurrencyById(currencyId);
+  const scheme = getDerivationScheme({
+    derivationMode,
+    currency,
+  });
+  const index = 0;
+  const freshAddressPath = runDerivationScheme(scheme, currency, {
+    account: index,
+    node: 0,
+    address: 0,
+  });
+  const account = {
+    type: "Account",
+    name:
+      currency.name +
+      " " +
+      (derivationMode || "legacy") +
+      " " +
+      shortAddressPreview(xpubOrAddress),
+    xpub: xpubOrAddress,
+    seedIdentifier: xpubOrAddress,
+    starred: true,
+    used: true,
+    swapHistory: [],
+    id,
+    derivationMode,
+    currency,
+    unit: currency.units[0],
+    index,
+    freshAddress: xpubOrAddress,
+    freshAddressPath,
+    freshAddresses: [],
+    creationDate: new Date(),
+    lastSyncDate: new Date(0),
+    blockHeight: 0,
+    balance: new BigNumber(0),
+    spendableBalance: new BigNumber(0),
+    operationsCount: 0,
+    operations: [],
+    pendingOperations: [],
+    balanceHistoryCache: emptyHistoryCache,
+  };
+  return account;
+}
+
+async function syncAccount(id) {
+  const account = inferAccount(id);
+  const bridge = getAccountBridge(account);
+  await bridgeCache.prepareCurrency(account.currency);
+  const syncConfig = {
+    paginationConfig: {},
+    blacklistedTokenIds: [],
+  };
+  const observable = bridge.sync(account, syncConfig);
+  const reduced = observable.pipe(reduce((a, f) => f(a), account));
+  const synced = await reduced.toPromise();
+  return synced;
+}
+
+function App() {
+  // synchronise account with an id that is input in a input text field
+  const [accountId, setAccountId] = useState("");
+  const [accountIdError, setAccountIdError] = useState("");
+
+  const [account, setAccount] = useState(null);
+  const [accountError, setAccountError] = useState("");
+
+  useEffect(() => {
```

Comment:
just a question, why can't we just merge these two "useEffect" hook into one?

Reply:
indeed it could be unified. they don't have exactly the same meaning and they could however be split into two parts, like one is doing the inference & the other is focusing on synching when it's valid.

fair feedback but not blocking this tools to land i think ;-)

Reply:
btw, i think a better design would actually be to offer some sort of `const accountId = useInferred(userInputText)` that is then fed into the last effect; i just did this very quick